В классе User добавлен статус для связи «дружба» между двумя пользова…

DmitriyDemintsev opened the pull request:
…телями, для статусов создан enum FriendsStatus

В классе Film добавлены свойства «жанр» и  «рейтинг», для них реализованы enum FilmRating и enum FilmGenre

Создана схема БД и примеры запросов к ней

DmitriyDemintsev committed:
В классе User добавлен статус для связи «дружба» между двумя пользователями, для статусов создан enum FriendsStatus

В классе Film добавлены свойства «жанр» и  «рейтинг», для них реализованы enum FilmRating и enum FilmGenre

Создана схема БД и примеры запросов к ней

DmitriyDemintsev committed:
В классе User добавлен статус для связи «дружба» между двумя пользователями, для статусов создан enum FriendsStatus

В классе Film добавлены свойства «жанр» и  «рейтинг», для них реализованы enum FilmRating и enum FilmGenre

Создана схема БД и примеры запросов к ней

исправлены найденные автопроверкий ошибки

DmitriyDemintsev committed:
В классе User добавлен статус для связи «дружба» между двумя пользователями, для статусов создан enum FriendsStatus

В классе Film добавлены свойства «жанр» и  «рейтинг», для них реализованы enum FilmRating и enum FilmGenre

Создана схема БД и примеры запросов к ней

исправлены найденные автопроверкий ошибки

DmitriyDemintsev committed:
В классе User добавлен статус для связи «дружба» между двумя пользователями, для статусов создан enum FriendsStatus

В классе Film добавлены свойства «жанр» и  «рейтинг», для них реализованы enum FilmRating и enum FilmGenre

Создана схема БД и примеры запросов к ней

исправлены найденные автопроверкий ошибки

DmitriyDemintsev committed:
В классе User добавлен статус для связи «дружба» между двумя пользователями, для статусов создан enum FriendsStatus

В классе Film добавлены свойства «жанр» и  «рейтинг», для них реализованы enum FilmRating и enum FilmGenre

Создана схема БД и примеры запросов к ней

исправлены найденные автопроверкий ошибки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

DmitriyDemintsev committed:
выполнена реализация в соответствии с ТЗ финального задания 11 спринта
исправлены замечания автопроверки

mikita95 commented on src/main/java/ru/yandex/practicum/filmorate/controller/GenreController.java:
```diff
@@ -0,0 +1,33 @@
+
+    private final FilmDbStorage filmDbStorage;
+
+    @Autowired
```
Можно не писать Autowired над конструкторами

mikita95 commented on src/main/java/ru/yandex/practicum/filmorate/dao/db/FilmDbStorage.java:
```diff
@@ -0,0 +1,208 @@
+    }
+
+    public void addLike(Film film, User user) {
+        jdbcTemplate.update("INSERT INTO likes (film_id, user_id) VALUES (?, ?)", film.getId(), user.getId());
```
Если выполнить эту операцию два раза, то будет ошибка, лучше использовать в данном случае оператор MERGE

mikita95 reviewed (changes requested):
Все сделано отлично. Оставил лишь небольшие замечания

DmitriyDemintsev committed:
учтены зачечания про @Autowired

исправлены запросы в методах addLike и addGenreToFilm в классе FilmDbStorage и addToFriends в классе UserDbStorage, реализация выполнена с использованием оператора MERGE

DmitriyDemintsev committed:
@Autowired

исправлены запросы в методах addLike и addGenreToFilm в классе FilmDbStorage и addToFriends в классе UserDbStorage, реализация выполнена с использованием оператора MERGE

DmitriyDemintsev committed:
испрпавлены виявленные при автопроверке git ошибки

учтены замечания про @Autowired

исправлены запросы в методах addLike и addGenreToFilm в классе FilmDbStorage и addToFriends в классе UserDbStorage, реализация выполнена с использованием оператора MERGE

mikita95 reviewed (approved):
Все отлично. Можно мержить пул-реквест
